homeassistant/components/seventeentrack/__init__.py

```
@@ -124,13 +129,73 @@ async def get_packages(call: ServiceCall) -> ServiceResponse:
             ]
         }
 
+    async def add_package(call: ServiceCall) -> ServiceResponse:
+        config_entry_id = call.data[ATTR_CONFIG_ENTRY_ID]
+        tracking_number = call.data.get(ATTR_TRACKING_NUMBER)
+        friendly_name = call.data.get(ATTR_INFO_TEXT, "")
+
+        config_entry: ConfigEntry | None = hass.config_entries.async_get_entry(
+            config_entry_id
+        )
+
+        if not config_entry:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="invalid_config_entry",
+                translation_placeholders={
+                    "config_entry_id": config_entry_id,
+                },
+            )
+        if config_entry.state != ConfigEntryState.LOADED:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="unloaded_config_entry",
+                translation_placeholders={
+                    "config_entry_id": config_entry.title,
+                },
+            )
+
+        seventeen_coordinator: SeventeenTrackCoordinator = hass.data[DOMAIN][
+            config_entry_id
+        ]
+
+        try:
+            new_package = await seventeen_coordinator.client.profile.add_package(
+                tracking_number=tracking_number, friendly_name=friendly_name
+            )
+        except SeventeenTrackError as err:
+            _LOGGER.error("Failed to add package %s: %s", tracking_number, err)
+            return None
```

Services shouldn't silently fail. Raise an `HomeAssistantError` instead.